Source/Core/Common/Deprecated.h

```diff
@@ -0,0 +1,12 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#ifdef _WIN32
```

Comment:
As this is compiler specific rather than OS specific, `_MSC_VER` would likely be more appropriate.

Reply:
Looks line Inline.h could use that, too.

Reply:
Should the \_\_attribute\_\_ decorator be only for GNU, or is it safe for it to always fall through to that option?

Reply:
Clang also supports [`always_inline`](https://clang.llvm.org/docs/AttributeReference.html#always-inline-force-inline) and [`deprecated`](https://clang.llvm.org/docs/AttributeReference.html#always-inline-force-inline).

Reply:
(however, I think clang does define `__GNUC__` so checking that way is fine)